docs/modules/sql/pages/user-defined-types.adoc

```diff
@@ -64,30 +57,39 @@ To create a new type, use the `CREATE TYPE` statement:
-    )
-----
-<1> `java` format requires the `javaClass` option.
-<2> `portable` format requires `portableFactoryId` and `portableClassId` and optionally `portableVersion`.
-<3> `compact` format requires `compactTypeName` - this is not the name of the created type, but rather internal name of the Compact record type, used internally by the Compact Serialization format.
+CREATE [OR REPLACE] TYPE [IF NOT EXISTS] MyTypeName [(
+    colName colType,
+    ...
+)] [OPTIONS (
+    'javaClass'='com.mypackage.MyJavaClass' <1>
+  | 'compactTypeName'='MyCompactRecordName' <2>
+  | 'portableFactoryId'='123', 'portableClassId'='456', ['portableVersion'='789'] <3>
+  | 'avroSchema'='{"type":"record","name":"myType","fields":[{"name":"colName","type":"colType"},...]}' <4>
+)]
+----
+<1> In `java` format, you can use the `javaClass` option to override the type class with a subclass. You must provide a concrete class for `INSERT` statements; this means that the `javaClass` option is mandatory when the original type class is abstract or interface.
+<2> In `portable` format, you can use the `portableFactoryId`, `portableClassId` and `portableVersion` options to specify a portable ID for the type. This is only effective when the type is used for `__key` or `this` fields. `portableVersion` is optional. However, components cannot be inherited from the parent, so portable ID components must be defined together. Incomplete definitions have the same effect as no definition.
+<3> In `compact` format, you can use the `compactTypeName` option to specify the Compact record name. If unspecified, it defaults to `<TypeName>CompactType`.
+<4> In `avro` format, you can use the `avroSchema` option to specify an inline Avro schema. This is only effective when the type is used for `__key` or `this` fields.
+
+NOTE: `EXTERNAL NAME` aliases are not supported for types, column names must have the same name as their corresponding Java class field or Portable/Compact/Avro schema field.
+[NOTE]
+====
+. `EXTERNAL_NAME` aliases  are not supported for types, column names must have the same name as their corresponding Java class field or Portable/Compact/Avro schema field.
+. You can mix options that belong to different formats. When you create a mapping that references your UDT, the relevant options are used and the others are ignored, which makes it possible to use a UDT in multiple mappings having different serialization formats.
+. To reference another type, you must provide the column list. Otherwise, in Java format, the column will be resolved as `OBJECT` type; in Portable, Compact and Avro formats, an exception will be thrown once the type is used in a mapping.
+====
+NOTE: You can mix options that belong to different formats. When you create a mapping that references your UDT, the relevant options are used and the others are ignored, which makes it possible to use a UDT in multiple mappings having different serialization formats.
```

Comment:
I think " it possible to use a UDT in multiple mappings having different serialization formats" is quite important feature (and significant improvement over previous approach) and should be mentioned in the overview. Do we have examples for that like read from kafka and write to IMap with the same UDT (OFC the syntax is not pretty as you have to recreate the object from fields)?

Reply:
Since there is no `format` option for UDTs, users would expect them to be format-agnostic. That's why, I think, we don't need to highlight this fact.

Reply:
but we say in the beginning that "portable, compact, java and avro formats are supported" - by analogy with mappings it might be assumed that UDT has also single assigned format. IMO worth mentioning this explicitly.

Reply:
this is documented explicitly only in note in "Creating Types"

Reply:
fef8c8c2